### ISSUE TITLE
Error Domain=com.google.fcm Code=505 "No APNS token specified before fetching FCM Token" UserInfo={NSLocalizedFailureReason=No APNS token specified before fetching FCM Token

### ISSUE THREAD

**Issue by jiangzhl** [firebase/firebase-ios-sdk#14195]
### Description

We used to run everything normally in ios17.2.1, but
this problem will occur when the application is run on a real device with ios18.1.1

### Reproducing the issue

This problem will occur when the app is run on a real device running iOS 18.1.1

### Firebase SDK Version

11.4

### Xcode Version

16.1

### Installation Method

Swift Package Manager

### Firebase Product(s)

Messaging

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Error Domain=com.google.fcm Code=505 "No APNS token specified before fetching FCM Token" UserInfo={NSLocalizedFailureReason=No APNS token specified before fetching FCM Token}
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by rizafran** [comment]
Thanks for reporting, @jiangzhl. I wasn't able to reproduce the issue (though I'm using a simulator v18.1). I just want to confirm if the issue is only reproducible specifically on iOS 18.1.1? Is the issue also occur in your simulator? 

Additionally, this error seems relevant to https://github.com/firebase/firebase-ios-sdk/issues/10679#issuecomment-1402776795 in which there was a change implemented on [v10.4.0](https://firebase.google.com/support/release-notes/ios#fcm_10) onwards.

**Comment by jiangzhl** [comment]
> 感谢您的举报，@jiangzhl。我无法重现该问题（尽管我使用的是模拟器 v18.1）。我只是想确认该问题是否仅在 iOS 18.1.1 上可重现？您的模拟器中是否也会出现此问题？
> 
> [此外，该错误似乎与#10679（评论）](https://github.com/firebase/firebase-ios-sdk/issues/10679#issuecomment-1402776795)有关，其中在[v10.4.0](https://firebase.google.com/support/release-notes/ios#fcm_10)及更高版本中实施了一项更改。

Please do this under the following conditions
1 Xcode 16.1
2 real mobile phone 18.1.1 system, not emulator

**Comment by jiangzhl** [comment]
> Thanks for reporting, @jiangzhl. I wasn't able to reproduce the issue (though I'm using a simulator v18.1). I just want to confirm if the issue is only reproducible specifically on iOS 18.1.1? Is the issue also occur in your simulator?
> 
> Additionally, this error seems relevant to [#10679 (comment)](https://github.com/firebase/firebase-ios-sdk/issues/10679#issuecomment-1402776795) in which there was a change implemented on [v10.4.0](https://firebase.google.com/support/release-notes/ios#fcm_10) onwards.
